tests/test_batch_edit.py

```
@@ -0,0 +1,320 @@
+                paragraph=refs[7].split("|")[0],
+            ),
+        ]
+
+        result = doc.batch_edit(ops)
+        assert len(result) == 3
+
+        vis = doc.get_visible_text()
+        assert "REPLACED" in vis
+        # Verify "item 5" was deleted (but [P05] marker remains in the paragraph)
+        assert "[P05]" in vis
+        p05_line = [line for line in vis.split("\n") if "[P05]" in line][0]
```

Ruff (0.15.7)

[warning] Prefer `next(...)` over single element slice

Replace with `next(...)`

(RUF015)